File: WebClientIS/WebClientIS/Models/RelCons.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace APIIS.Models
{
    public class RelCons
    {
        public DateTime DataRel { get; set; }

        public int IdU { get; set; }

        public int IdC { get; set; }

        public RelCons()
        {
            DataRel = new DateTime();
            IdU = 0;
            IdC = 0;
        }


        public List<RelCons> GetRelCons(int idu)
        {
            List<RelCons> lista = new List<RelCons>();

            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/RelCons/" + idu.ToString());

            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            HttpResponseMessage response = client.GetAsync("").Result;

            if (response.IsSuccessStatusCode)
            {
                var products = response.Content.ReadAsStringAsync().Result;

                var array = JArray.Parse(products);

                foreach (var obj in array)
                {
                    try
                    {
                        lista.Add(obj.ToObject<RelCons>());
                    }
                    catch (Exception ex)
                    {
                        return new List<RelCons>();
                    }
                }

            }

            return lista;
        }

        public bool CreateRelCons(RelCons relcons)
        {
            HttpClient client = new HttpClient();

            var values = new Dictionary<string, string>
            {
                { "DataRel", relcons.DataRel.ToString() },
                {"IdU",relcons.IdU.ToString() },
                {"IdC",relcons.IdC.ToString() }


            };

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/RelCons");

            var jsonString = JsonConvert.SerializeObject(values);
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PostAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: MobileClient/sampleproject/lib/Models/RelCons.dart
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert' show json;
import 'package:scoped_model/scoped_model.dart';

class RelCons extends Model{

  DateTime datarel;
  int idu;
  int idc;

  RelCons({DateTime datarele,int iduu,int idcc})
  {
    this.datarel = datarele;
    this.idu = iduu;
    this.idc = idcc;
  }

  factory RelCons.fromJson(Map<String,dynamic> json){

    return RelCons(
      datarele:DateTime.parse(json['DataRel']),
      iduu: json['idU'] as int,
      idcc: json['idC'] as int
      
    );

  }

  Future<List<RelCons>> getrelcons() async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/RelCons"),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<RelCons> listau = new List<RelCons>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(RelCons.fromJson(lista[i]));
  

  return listau;
}

Future<List<RelCons>> getrelconsidu(int idu) async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/RelCons/"+idu.toString()),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<RelCons> listau = new List<RelCons>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(RelCons.fromJson(lista[i]));
  

  return listau;
}

Future<int> createRelCons(RelCons relCons) async {

  var url ='http://6f85a52b4f05.ngrok.io/api/RelCons';
  var body = json.encode(<String, String>{
      'DataRel': relCons.datarel.toString(),
      'IdU':relCons.idu.toString(),
      'IdC':relCons.idc.toString()
      
    });

  print(body);



   http.Response response = await http.post(url,
      headers: {"Content-Type": "application/json"},
      body: body
  );

  return response.statusCode;
}

}